Better settings menu + basic debugging features

memedmusic opened the pull request:
Just redesigned the configure saladbind interface + added basic debugging features such as: clear cache, clear settings, remove miners.
![Capture3](https://user-images.githubusercontent.com/55370047/131458510-c7845327-d25c-4119-a92d-ee6f38ab8982.PNG)
![Capture1](https://user-images.githubusercontent.com/55370047/131458508-d3130f3a-f86d-4cc6-b98a-e6744c37db6d.PNG)
![Capture](https://user-images.githubusercontent.com/55370047/131458501-a15b50c2-e12c-4d37-ac64-5d520c8fed14.PNG)

Vukkyy reviewed (commented):
is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol

memedmusic commented:
> is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol

Nah, no reason lol. Would you prefer a different colour?

Vukkyy commented:
> > is there any particular reason you have made all the debug options orange? looks kind of ugly and gives me cli+ vibes lol
> 
> Nah, no reason lol. Would you prefer a different colour?

how abut not setting one? white is... good enough, y'know

Vukkyy commented on index.js:
```diff
@@ -64,7 +68,10 @@ if (process.argv[2]) {
-				require("./setup").run(false);
+				require("./setup").first(false, false, true);
+			} else if (config.version == undefined) {
+				console.log("Looks like something isn't quite right with your settings. Lets fix them :)\n");
```
```suggestion
				console.log("Looks like something isn't quite right with your settings. Let's fix them :)\n");
```

Vukkyy commented on setup.js:
```diff
@@ -154,25 +165,174 @@ async function continueSetup(clear) {
+	}
+
+}
+
+
+
```
why do you like new lines so much lol

memedmusic replied:
heh

vtheskeleton reviewed (changes requested):
After entering worker id the program exits
Also the orange looks kinda ugly, we should try to keep things white and only make notes orange (like in the miner selection screen where compatibility is noted in orange)

vtheskeleton commented:
Also if it fails to find your worker id from logs it just goes to the menu anyway (even though there is a notice about it), still kinda bad design as the user may not realize something is wrong!

memedmusic commented:
okei, I'll go change these thingies

memedmusic commented:
Made colours better, removed a few extra blank lines, fixed spelling errors, fixed auto logs bug
![Capture](https://user-images.githubusercontent.com/55370047/131462241-e00522dc-b247-4644-a4d2-e75abd49526c.PNG)
![Capture2](https://user-images.githubusercontent.com/55370047/131462246-fae40a5d-9e4e-4f88-8e86-e4286d54cfb9.PNG)

Vukkyy reviewed (changes requested):
why does settings-main use `message:` in inquirer as its heading while the others console.log("settings - stuff") and do `message: choose an action`?

(i prefer the latter)

TheBlueBurger commented on setup.js:
```diff
@@ -88,28 +108,22 @@ async function continueSetup(clear) {
 			if (rigID) rigID = rigID.split(": ")[1];
 			return rigID;
 		}
-		let rigID = getIDFromLogs("main.log") ?? getIDFromLogs("main.old.log")
```
Why did you remove let

memedmusic replied:
The save function is seperate from the minerDetails function, and using let doesnt expose the rigID variable to the whole script while removing let does

memedmusic commented:
> why does settings-main use `message:` in inquirer as its heading while the others `console.log("settings - stuff")` and do `message: choose an action`?
> 
> (i prefer the latter, though it would be nice to not have the Settings - prefix, as... well, it already says Configure SaladBind, of course I'm in the settings lol)

Idrk lmao, I did it because, well.. i did. I guess I'll remove the console.log headers, remove the settings prefix, and change the 'message:' headers around

memedmusic commented:
Changed the headers around a bit

Vukkyy commented on setup.js:
```diff
@@ -248,11 +246,10 @@ async function first() {
 		type: 'list',
 		name: "debug",
-		message: chalk.bold.white("Choose an action below"),
+		message: chalk.bold.white("Debugging tools"),
```
i said the latter, not the former... :p

memedmusic replied:
oh, right. mb

vtheskeleton commented:
Request a new review when you fix all the issues

Vukkyy commented:
unfortunately this seems to be heavily broken at the point of no return when i tried it out, but i appreciate you working on it! :)
